balancer_wrapper.go

```diff
@@ -352,13 +350,23 @@ func (acbw *acBalancerWrapper) GetOrBuildProducer(pb balancer.ProducerBuilder) (
 	// and delete the refCountedProducer from the map if the total reference
 	// count goes to zero.
 	unref := func() {
-		acbw.mu.Lock()
+		acbw.producersMu.Lock()
 		pData.refs--
```

Comment:
Could this become negative if the LB policy calls `close()` after the subchannel state has changed? 

We can check if `acbw.producers[pb] != pData` and exit early to avoid this.

Reply:
It can, and I hadn't actually considered that.  In this case, it isn't actually a problem, since refs will become negative and never call `close` again.  So I think I could just add a comment, or check the `producers` map as you suggested, or pre-check `refs==0`, or clear the `close` field or something else like that.

I'm inclined to just add a comment.  LMK if you disagree.

Reply:
Sounds good.